src/managers/auth.py

```diff
@@ -25,85 +27,148 @@ class Acl:
 
     username: str
     operation: Operation
-    resource_type: Resource
-    resource_name: str
+    resource: str
 
     @property
-    def resource_str(self) -> str:
-        """Returns the full resource string."""
-        return f"{self.resource_type}:{self.resource_name}"
-
-    @property
-    def full_acl_repr(self) -> dict:
+    def acl_repr(self) -> dict[str, str]:
         """Returns the dict with an ACL representation."""
         return {
             "username": self.username,
             "operation": self.operation,
-            "resource": self.resource_str,
+            "resource": self.resource,
         }
 
 
+@dataclass
+class UserCredentials:
+    """Object for representing credentials of a Karapace user."""
+
+    username: str
+    algorithm: Algorithm
+    salt: str
+    password_hash: str
+
+    @property
+    def creds_repr(self) -> dict[str, str]:
```

Comment:
**todo:** You can remove this method, and use `dataclasses.asdict` wherever you need it

```python
from dataclasses import dataclass, asdict

@dataclass
class X:
    foo: str

x = X(foo="hello")
print(asdict(x))

>>> {'foo': 'hello'}
```

Reply:
TIL, thanks!